### DISCUSSION TITLE
Adding A Plugin

### DISCUSSION THREAD

**Discussion by C4E5G3** [NotAShelf/nvf#391] [Q&A] [answered]
In my plugins.nix, I am trying to add the base16-schemes plugin, in order to use the base16-classic-dark theme. However, it is generating an error.
```
module 'base16-schemes' not found:                                                                                              
        no field package.preload['base16-schemes']    
```

 Any thoughts? 
```
 
{ pkgs, ... }: {
  
  programs.nvf.settings = {    
    vim.extraPlugins = with pkgs; {
    base16-schemes = {
      package = base16-schemes;
      setup = ''
        require('base16-schemes').setup {
          vim.cmd( [[colorscheme base16-classic-dark]] )
        }
      '';
    };

  };
};

```

**Comment by NotAShelf** [comment, marked as answer]
`vim.extraPlugins` only accepts *(n)vim plugins* in the package field. You cannot use a package that's not a Vim plugin, since it'd be incorrectly added to Neovim's own runtime path. Additionally, base16-schemes in nixpkgs is not a neovim plugin - `require('base16-schemes').setup` would not work.

We have #389 in the works for adding base16 support, or you could set up a Neovim plugin that supports base16 theming directly.

**Reply by C4E5G3** [reply]
Thanks Raf. I will wait for Bloxx to complete his work I think
